module/VuFind/src/VuFind/Sitemap/Generator.php

```diff
@@ -331,12 +344,21 @@ protected function getIdsFromBackendUsingCursorMark(Backend $backend, $cursorMar
         }
         $prevCursorMark = $cursorMark;
 
+        // Set up the record factory on first call. We use a very simple factory
+        // since performance is important and we only need the identifier.
+        if ('*' === $cursorMark) {
```

Comment:
Would it be cleaner to have a `configureBackend` method that gets called at the top of `generateForBackend` and does this work conditionally based on `$this->retrievalMode` rather than basing it on the cursor mark, which seems less clear and more potentially error prone?

Reply:
Yes. I made it setupBackend.